products/sentinel_one.py

```diff
@@ -461,17 +471,28 @@ def _process_queries(self):
 
             for tag, queries in self._queries.items():
                 for query in queries:
-                    if query.operator in ('contains', 'containscis', 'contains anycase'):
+                    if query.operator in ('contains', 'containscis', 'contains anycase') and not re.findall(r'(?:\" AND)', query.search_value):
                         key = (query.operator, query.parameter)
                         if key not in combined_queries:
                             combined_queries[key] = list()
 
                         combined_queries[key].append((tag, query.search_value))
                     elif query.full_query is not None:
                         query_text.append((tag, query.full_query))
-                    else:
-                        full_query = query.parameter + ' ' + query.operator + ' ' + query.search_value
+                    elif query.operator == 'raw' and query.parameter == 'query':
+                        temp_search_value = query.search_value.lstrip('\"')
+                        if temp_search_value.endswith(')"'):
+                            temp_search_value = temp_search_value.rstrip('\"')
+                        full_query = f'({temp_search_value})'
```

Comment:
We only _have_ to check either the operator or parameter but it's not necessary to do both since one implies the other. Also, no need to format the `query.search_value` since we adjusted how that is stored up on line 431.
 
I'm not able to annotate this but line 487 will need to be deleted.

```suggestion
                    elif query.operator == 'raw':
                        query_text.append((tag, query.search_value))
```

Reply:
this is still needed due to how json formats things. you need to escape double quotes and backslashes. without these lines, the beginning and end quotes get added causing an error. I have removed the double check in the initial if statement.

Reply:
added this query to the test.json file. this is a scenario where the quotes can cause an issue.
```
"mixed_operator_query": {
        "query":["TgtProcName = \"chrome.exe\" AND (DnsRequest = \"gmail.com\" OR DnsRequest = \"outlook.com\")"]
    }
```

Reply:
latest commit has fixes for this and removes the stripping of quotes logic. I added a check to only group queries together if more than one exists otherwise set the terms to the search_value this fixed the quotes issue.